### ISSUE TITLE
Negative integers

### ISSUE THREAD

**Issue by moctes** [nakagami/efirebirdsql#19]
Hello,

I am using efirebirdsql through firebirdex, I have a domain defined in the database like this:

`CREATE DOMAIN D_ID AS BIGINT NOT NULL;`

So basically the fields based on this domain are BIGINT I guess it doesn't actually matter the domain but the base type, when I try to insert a record with a negative value (for example -1), for one of these fields I get the following error:

 ```
* 1st argument: out of range

    (stdlib 6.0) :binary.encode_unsigned(-1, :big)
    (efirebirdsql 0.9.9) /home/myuser/myproject/deps/efirebirdsql/src/efirebirdsql_conv.erl:24: :efirebirdsql_conv.byte4/2
    (efirebirdsql 0.9.9) /home/myuser/myproject/deps/efirebirdsql/src/efirebirdsql_conv.erl:178: :efirebirdsql_conv.param_to_blr/2
    (efirebirdsql 0.9.9) /home/myuser/myproject/deps/efirebirdsql/src/efirebirdsql_conv.erl:247: :efirebirdsql_conv.params_to_blr/5
    (efirebirdsql 0.9.9) /home/myuser/myproject/deps/efirebirdsql/src/efirebirdsql_conv.erl:262: :efirebirdsql_conv.params_to_blr/3
    (efirebirdsql 0.9.9) /home/myuser/myproject/deps/efirebirdsql/src/efirebirdsql_op.erl:346: :efirebirdsql_op.op_execute2/3
    (efirebirdsql 0.9.9) /home/myuser/myproject/deps/efirebirdsql/src/efirebirdsql_protocol.erl:232: :efirebirdsql_protocol.execute/4
    (firebirdex 0.3.11) lib/firebirdex/connection.ex:95: Firebirdex.Connection.handle_execute/4
    (db_connection 2.7.0) lib/db_connection/holder.ex:354: DBConnection.Holder.holder_apply/4
    (db_connection 2.7.0) lib/db_connection.ex:1558: DBConnection.run_execute/5
    (db_connection 2.7.0) lib/db_connection.ex:1653: DBConnection.run/6
    (db_connection 2.7.0) lib/db_connection.ex:772: DBConnection.parsed_prepare_execute/5
    (db_connection 2.7.0) lib/db_connection.ex:764: DBConnection.prepare_execute/4
    (firebirdex 0.3.11) lib/firebirdex.ex:23: Firebirdex.query/4
    (ecto_sql 3.11.3) lib/ecto/adapters/sql.ex:1108: Ecto.Adapters.SQL.struct/10
    (ecto 3.11.2) lib/ecto/repo/schema.ex:775: Ecto.Repo.Schema.apply/4
    (ecto 3.11.2) lib/ecto/repo/schema.ex:377: anonymous fn/15 in Ecto.Repo.Schema.do_insert/4
    (ecto 3.11.2) lib/ecto/repo/schema.ex:273: Ecto.Repo.Schema.insert!/4
```

Is it possible to change the driver in such a way that it accepts negative numbers for integer type fields? It's something we do in many tables with many fields

Thank you for taking the time to maintain this project, I hope this change can be useful to me and others

**Comment by nakagami** [comment]
Thanks,

Fixed and released as 0.9.10

**Comment by moctes** [comment]
Awesome !

We had not had the opportunity to update the dependencies, now we can actually save negative integers :)

Thank you !!
